internal/docker/deployer.go

```diff
@@ -301,89 +338,21 @@ func deployImage(
 		log.Printf("%s: Started container %s", contextStr, containerID)
 	}
 
-	// We need to hammer the inspect endpoint until the ports show up, they don't appear immediately.
-	var inspect types.ContainerJSON
-	var baseURL, fedBaseURL string
-	inspectStartTime := time.Now()
-	for time.Since(inspectStartTime) < time.Second {
-		inspect, err = docker.ContainerInspect(ctx, containerID)
-		if err != nil {
-			return stubDeployment, err
-		}
-		if inspect.State != nil && !inspect.State.Running {
-			// the container exited, bail out with a container ID for logs
-			return stubDeployment, fmt.Errorf("container is not running, state=%v", inspect.State.Status)
-		}
-		baseURL, fedBaseURL, err = endpoints(inspect.NetworkSettings.Ports, 8008, 8448)
-		if err == nil {
-			break
-		}
-	}
+	baseURL, fedBaseURL, err := waitForPorts(ctx, docker, containerID)
 	if err != nil {
 		return stubDeployment, fmt.Errorf("%s : image %s : %w", contextStr, imageID, err)
 	}
+	inspect, err := docker.ContainerInspect(ctx, containerID)
+	if err != nil {
```

Comment:
I don't think this will ever be hit? `waitForPorts` does `inspect, err = docker.ContainerInspect(ctx, containerID)` initially and then bails if there is an `err`. In that case, we would've returned already on `:342`.